### ISSUE TITLE
Idle connection shutdown cannot be easily detected by client

### ISSUE THREAD

**Issue by hectorj** [aerospike/aerospike-server#156]
Hi!

I was having some issues with EOF errors on some write commands, using the Go client and aerospike-server 1.9.1.1

My team and I tracked it down to an issue with idle connections timeouts: our `IdleTimeout` client-side was bigger than the server-side `proto-fd-idle-ms`.

You can see there that the Java client advise against it: https://github.com/aerospike/aerospike-client-java/blob/66a0899462b89e85f8cca03e0ecd2d05d6035386/client/src/com/aerospike/client/policy/ClientPolicy.java#L65-L67

But I tried to understand the issue a bit more, and it seems to come down to this: http://stackoverflow.com/q/11436013/1685538

When the server shuts down the socket of an idle connection, clients do not detect it on the first write, but only after, when trying to read the response. (there is then an issue with how the Go client handle that read error, but I'll create another issue in the appropriate repository for that)

The Stack Overflow answer suggests setting the socket's linger option to 0 before **closing** it, which allows the client to know immediately that the connection was reset.

Is that something that could be done in aerospike-server? I'm not sure why a `shutdown` `RW` is done instead of a `close`, I don't have enough experience in this domain.
